spring-cloud-stream-metrics/src/main/java/org/springframework/cloud/stream/metrics/ApplicationMetricsProperties.java

```diff
@@ -99,6 +120,29 @@ public void setProperties(String[] properties) {
 	private Map<String, Object> exportProperties = new HashMap<>();
 
 	public Map<String, Object> getExportProperties() {
+		this.exportProperties.clear();
```

Comment:
Wouldn't this  cause the `exportProperties` array to be computed each time we send out a metrics message? That would be very inefficient, as the list doesn't change between refresh operations. That was the main reason why it's been done once via `onApplicationEvent` - based on the discussion that @jvalkeal and I had, that seems to have some race condition issue, but we should probably make sure that this is cached rather than computed every time.